lib/libzfs/libzfs_import.c

```diff
@@ -820,12 +820,31 @@ pool_active(libzfs_handle_t *hdl, const char *name, uint64_t guid,
 	return (0);
 }
 
+boolean_t
+zfs_forceimport_required(nvlist_t *config)
+{
+	uint64_t state;
+	uint64_t hostid = 0;
+	unsigned long system_hostid = get_system_hostid();
+
+	state = fnvlist_lookup_uint64(config, ZPOOL_CONFIG_POOL_STATE);
+	(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID,
+	    &hostid);
```

Comment:
My understanding is that ZPOOL_CONFIG_HOSTID is only present if the last time the pool was exported, the hostid was set.  So it's not an error if the key is missing.

I cannot return early, because if the hostid was not set before, but is set now, and the pool state is POOL_STATE_ACTIVE, then I need to perform the activity test.  I cannot tell the difference between the hostid changing on this host, vs. importing the pool on a new host.